### PR TITLE
[PLT-1506] Added Ruff linting to SDK

### PR DESCRIPTION
**Pull request by Gabefire**
# Description
* This adds in the Ruff Linter that is the same package as the Ruff formator we now use
* if this merges SDK devs will HAVE to use Ruff for linting this is the same tool for formatting
* Note mypy type checker is separate from this linter 
* Using default rules besides Flake which is showing 300+ violations right now mostly unused imports. I also ignore one rule for bare excepts since that is a rule we break a lot
* View this doc for all [rules](https://docs.astral.sh/ruff/rules/)

### CODE REVIEW

**Comment by vbrodsky** [comment]
@Gabefire is it linting? or removing ND deserialization? Or both?

File: libs/labelbox/src/labelbox/data/serialization/ndjson/base.py [hunk 1]
```diff
@@ -8,18 +8,6 @@
 
 from ....annotated_types import Cuid
 
-subclass_registry = {}
-
-
-class _SubclassRegistryBase(BaseModel):
```
Comment (vbrodsky) [inline libs/labelbox/src/labelbox/data/serialization/ndjson/base.py]: Yeah!!!

**Review by vbrodsky — APPROVED** [review]
Great job

**Review by vbrodsky — CHANGES_REQUESTED** [review]
Had to remove approval due to failing tests for p3.12 version

File: libs/labelbox/pyproject.toml [hunk 2]
```diff
@@ -74,6 +74,10 @@ dev-dependencies = [
 [tool.ruff]
 line-length = 80
 
+[tool.ruff.lint]
+ignore = ["F", "E722"]
+exclude = ["**/__init__.py"]
```
Comment (Gabefire) [inline libs/labelbox/pyproject.toml:79]: This needs to be excluded for future rules (no unused imports)

File: libs/labelbox/pyproject.toml [hunk 3]
```diff
@@ -74,6 +74,10 @@ dev-dependencies = [
 [tool.ruff]
 line-length = 80
 
+[tool.ruff.lint]
+ignore = ["F", "E722"]
```
Comment (Gabefire) [inline libs/labelbox/pyproject.toml:78]: I have PyFlake and a bareexcept rule ignored here. Later we should look at not ignoring these rules and possiblly include more then just the defaults.

File: libs/labelbox/pyproject.toml [hunk 4]
```diff
@@ -89,9 +93,10 @@ unit = "pytest tests/unit"
 # LABELBOX_TEST_BASE_URL="http://host.docker.internal:8080" \
 integration = { cmd = "pytest tests/integration" }
 data = { cmd = "pytest tests/data" }
+rye-lint = "rye lint"
 rye-fmt-check = "rye fmt --check"
 mypy-lint = "mypy src --pretty --show-error-codes --non-interactive --install-types"
-lint = { chain = ["mypy-lint", "rye-fmt-check"] }
+lint = { chain = ["rye-fmt-check", "mypy-lint", "rye-lint"] }
```
Comment (vbrodsky) [inline libs/labelbox/pyproject.toml:99]: do we still need mypy here?

**Review by vbrodsky — APPROVED** [review]

File: libs/labelbox/pyproject.toml (hunk 4 above)
Comment (Gabefire) [reply libs/labelbox/pyproject.toml:99]: Yea I think so Mypy is a [type checker](https://docs.astral.sh/ruff/faq/#how-does-ruff-compare-to-mypy-or-pyright-or-pyre)

**Pull request merged**